### PR TITLE
Update command_line_interface.md

### PR DESCRIPTION
**Pull request by justinxzhao**

### CODE REVIEW

File: docs/user_guide/command_line_interface.md [hunk 1]
```diff
@@ -172,22 +173,25 @@ the training process gets interrupted somehow.
...
 This is useful for reproducibility.
-Be aware that due to asynchronicity in the TensorFlow GPU execution, when training on GPU results may not be reproducible.
+
+Be aware that due to asynchronicity in the Torch's GPU execution, when training on GPU results may not be reproducible.
 
 You can manage which GPUs on your machine are used with the `--gpus` argument, which accepts a string identical to the format of `CUDA_VISIBLE_DEVICES` environment variable, namely a list of integers separated by comma.
 You can also specify the amount of GPU memory that will be initially assigned to TensorFlow with `--gpu_memory_limit`.
 By default all of memory is allocated.
-If less than all of memory is allcoated, TensorFlow will need more GPU memory it will try to increase this amount.
+If less than all of memory is allocated, TensorFlow will need more GPU memory it will try to increase this amount.
```
Comment (dantreiman) [inline docs/user_guide/command_line_interface.md:190]: s/Tensorflow/Torch

File: docs/user_guide/command_line_interface.md [hunk 2]
```diff
@@ -172,22 +173,25 @@ the training process gets interrupted somehow.
...
 
-You can specify a random seed to be used by the python environment, python random package, numpy and TensorFlow with the `--random_seed` argument.
+You can specify a random seed to be used by the python environment, python random package, numpy and Torch with the `--random_seed` argument.
 This is useful for reproducibility.
-Be aware that due to asynchronicity in the TensorFlow GPU execution, when training on GPU results may not be reproducible.
+
+Be aware that due to asynchronicity in the Torch's GPU execution, when training on GPU results may not be reproducible.
 
 You can manage which GPUs on your machine are used with the `--gpus` argument, which accepts a string identical to the format of `CUDA_VISIBLE_DEVICES` environment variable, namely a list of integers separated by comma.
 You can also specify the amount of GPU memory that will be initially assigned to TensorFlow with `--gpu_memory_limit`.
```
Comment (dantreiman) [inline docs/user_guide/command_line_interface.md:188]: the maximum amount of GPU memory which will be allocated per device

File: docs/user_guide/command_line_interface.md [hunk 3]
```diff
@@ -738,17 +742,56 @@ optional arguments:
...
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -d DATASET, --dataset DATASET
+                        input data file path
+  -t TARGET, --target TARGET
+                        target(s) to predict as output features of the model
+  --time_limit_s TIME_LIMIT_S
+                        time limit to train the model in seconds when using hyperopt
+  --tune_for_memory TUNE_FOR_MEMORY
```
Comment (dantreiman) [inline docs/user_guide/command_line_interface.md:760]: --tune_for_memory is a boolean flag afaik. (true/false)

File: docs/user_guide/command_line_interface.md [hunk 4]
```diff
@@ -738,17 +742,56 @@ optional arguments:
...
+                        refine hyperopt search space based on available host / GPU memory
+  --hyperopt HYPEROPT   include automl hyperopt config
```
Comment (dantreiman) [inline docs/user_guide/command_line_interface.md:762]: Also a bool

File: docs/user_guide/command_line_interface.md [hunk 5]
```diff
@@ -738,17 +742,56 @@ optional arguments:
...
+  --random_seed RANDOM_SEED
+                        seed for random number generators used in hyperopt to improve repeatability
+  --use_reference_config USE_REFERENCE_CONFIG
```
Comment (dantreiman) [inline docs/user_guide/command_line_interface.md:765]: One more bool parameter

**Review by dantreiman — CHANGES_REQUESTED** [review]
Looks good overall, added a few notes

File: docs/user_guide/command_line_interface.md (hunk 2 above)
Comment (justinxzhao) [reply docs/user_guide/command_line_interface.md]: Done.

File: docs/user_guide/command_line_interface.md (hunk 1 above)
Comment (justinxzhao) [reply docs/user_guide/command_line_interface.md]: Good catch! Done.

File: docs/user_guide/command_line_interface.md (hunk 3 above)
Comment (justinxzhao) [reply docs/user_guide/command_line_interface.md:760]: You're right.

This is automatically generated from argparse, so if we want to update the strings for the boolean flags here, I should also update it in `cli.py`.

Perhaps:

```
--tune_for_memory TUNE_FOR_MEMORY
                        (bool) refine hyperopt search space based on available host / GPU memory
```

or:

```
--tune_for_memory TUNE_FOR_MEMORY
                        (true/false) refine hyperopt search space based on available host / GPU memory
```

?

**Review by dantreiman — APPROVED** [review]

**Pull request merged**